src/tls_stream.rs

```diff
@@ -617,16 +617,12 @@ impl<S> TlsStream<S>
     }
 
     fn write_out(&mut self) -> io::Result<usize> {
-        let mut out = 0;
-        while self.out_buf.position() as usize != self.out_buf.get_ref().len() {
-            let position = self.out_buf.position() as usize;
-            let nwritten = try!(self.stream.write(&self.out_buf.get_ref()[position..]));
-            out += nwritten;
-            self.out_buf.set_position((position + nwritten) as u64);
-        }
-
-        Ok(out)
-    }
+        let mut nwritten = 0;
+		let position = self.out_buf.position() as usize;
+		nwritten += self.stream.write(&self.out_buf.get_ref()[position..])?;
```

Comment:
Ah if other code expects write_out to do that then I guess we have to change it back. Like I said above this was specifically changed to abide by `A call to write represents at most one attempt to write to any wrapped object.` . Maybe we can change the other code that depends on that behavoir? I dont know the surrounding code that well.